### PR TITLE
Task#23 - Feat: Shopping cart component

### LINKED ISSUE #90: User Story#1: Display Cart Summary Sidebar
> As a user, I want a button that, when clicked, triggers a slide-in sidebar from the right, occupying one-third of the screen. This sidebar should display a list of products that have been added to the cart.

- #91 

- [x] Design a prominent button labeled "Cart" and place it in a visible location on the screen.
- [x] Define the appearance of the slide-in sidebar, including its dimensions, color scheme, and transition animation.
- [x] Implement the functionality to display the sidebar when the "Cart" button is clicked.

- #93

- [x] Retrieve the list of products that have been added to the cart and display them in the sidebar.
- [x] Scrolls appear if the product is too much
- [x] Show UI product names, quantities, individual prices, and a total cost calculation.
- [x] Show a button so the user can Proceed to Checkout

- Task#3: Test Cart Summary Sidebar

- [x] Test the interaction of the "Cart" button and the slide-in sidebar.
- [x] Ensure the sidebar functions smoothly on various devices and screen sizes.
- [x] Verify that the cart contents are accurately displayed and updated.

### LINKED ISSUE #91: Task#23 - DevTask: Create Cart Summary Button and Sidebar
- #89 
  - #90  

## OVERVIEW

Buttons: 

- Create a button with cart icon
- Add chip show summanry Quantity product in cart

Slidebars:
- Create a slide bard with 1/3 size on desktop and full screen on mobile screen

## DEMO 
[Link](https://next-commerce-v2.vercel.app/matte-black-magic-mug)
- Button summary
image
- Slidebar

**Component:**

image

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #91 

## OVERVIEW
- Add cart image to the data
- Create ShoppingCart components
- Update props for Popover components
- Update class style for Button components

## CHECKLIST
- [x] Fixed scrolling when displaying the cart
- [x] Close the cart when clicking on the background
- [x] Re-update the user interface of the cart card
- [x] Remove unnecessary animation for the shopping cart
- [x] Update the button position of the cart

## DEMO

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/8cecb942-6c91-4013-9ed2-0d12997b394d

### CODE REVIEW

File: practice-two/src/components/common/Popover.jsx [hunk 1]
```diff
@@ -9,19 +9,25 @@ import Button from './Button/Button';
...
   isOpen = false,
   onClosePopover = () => {},
   closeButton = true,
+  closeButtonContent = 'Close',
+  closeButtonStyle = '',
   children,
   popoverClassName,
   arrowPopover = true,
   isFixed = false,
+  fixedOverlayClassName,
```
Comment (phattruong-agilityio) [inline practice-two/src/components/common/Popover.jsx]: Not a good solution. it's all about className and style, it's should not stuck in only special case

File: practice-two/src/components/ShoppingCart.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  const handleButtonClick = () => {
+    setIsPopoverOpen((prevState) => !prevState);
+    document.body.classList.remove('overflow-hidden');
+  };
+
+  return (
+    <div>
+      <Button customVariant="w-11 h-11 relative" onClick={handleButtonClick}>
+        <img src={ICON.cart} alt="filter-icon" className="w-6" />
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ShoppingCart.jsx]: check `alt`

File: practice-two/src/components/ShoppingCart.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          arrowPopover={false}
+        >
+          <div className="w-full">
+            <div className="inline-block">
+              <Typography level={2} size="xl" variant="soild">
+                My Cart
+              </Typography>
+            </div>
+            <div className="mt-20 flex w-full flex-col items-center justify-center overflow-hidden">
+              <img src={ICON.cart} alt="filter-icon" className="w-24" />
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ShoppingCart.jsx]: `alt`

File: practice-two/src/components/layout/FilterCategoryLayout.jsx [hunk 4]
```diff
@@ -10,6 +10,7 @@ import Button from '../common/Button/Button';
 // Data
 import BRAND from '../../constants/brand';
 import { PRICES, CATEGORIES } from '../../constants/data';
+import ShoppingCart from '../ShoppingCart';
```
Comment (phattruong-agilityio) [inline practice-two/src/components/layout/FilterCategoryLayout.jsx]: It's a data?

File: practice-two/src/components/ShoppingCart.jsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import Popover from './common/Popover';
+import Typography from './common/Typography';
+
+// Data
+import { ICON } from '../constants/data';
+
+/**
+ * ShoppingCart Component
+ *
+ * A popover component for displaying cart products.
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ShoppingCart.jsx]: Wrong! It's not a popover. It's sidebar.  and it's should not relative to 'Popover' component

File: practice-two/src/components/ShoppingCart.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-two/src/components/ShoppingCart.jsx]: Yes, udpated

File: practice-two/src/components/ShoppingCart.jsx (hunk 3 above)
Comment (TheHienDuong) [reply practice-two/src/components/ShoppingCart.jsx]: Yes, Updated

File: practice-two/src/components/layout/FilterCategoryLayout.jsx (hunk 4 above)
Comment (TheHienDuong) [reply practice-two/src/components/layout/FilterCategoryLayout.jsx]: Yes, Updated

File: practice-two/src/components/ShoppingCart.jsx (hunk 5 above)
Comment (TheHienDuong) [reply practice-two/src/components/ShoppingCart.jsx]: Done, Updated by Slider components

File: practice-two/src/components/common/Popover.jsx (hunk 1 above)
Comment (TheHienDuong) [reply practice-two/src/components/common/Popover.jsx]: Yes, Im change name fixedOVerlayClassName into customClassNames

**Comment by phattruong-agilityio** [comment]
What is the different between these props. Check your naming plz

image

**Comment by phattruong-agilityio** [comment]
Why Button is inside a folder but popover is not?
image

File: practice-two/src/components/SliderBar.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      {isFixed && (
+        <button
+          className={`fixed inset-0 bg-opacity-70 backdrop-filter backdrop-blur-lg flex items-center justify-center bg-gray-800 ${customClassNames}`}
+          onClick={onCloseSliderBar}
+          onKeyDown={onCloseSliderBar}
+        >
+          {/* Content goes here */}
+        </button>
+      )}
+      <div className="z-30">
```
Comment (phattruong-agilityio) [inline practice-two/src/components/SliderBar.jsx]: why using `<div>` in `sidebar`. Check your HTML tag here

File: practice-two/src/components/SliderBar.jsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+SliderBar.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
+  onCloseSliderBar: PropTypes.func,
+  closeButton: PropTypes.bool,
+  closeButtonContent: PropTypes.string,
+  closeButtonStyle: PropTypes.string,
+  sliderBarClassName: PropTypes.string,
+  isFixed: PropTypes.bool,
+  customClassNames: PropTypes.string, // Changed from fixedOverlayClassName
```
Comment (phattruong-agilityio) [inline practice-two/src/components/SliderBar.jsx]: Does this comment is helpful? `Changed from fixedOverlayClassName`

File: practice-two/src/components/SliderBar.jsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+  return isOpen ? (
+    <>
+      {isFixed && (
+        <button
+          className={`fixed inset-0 bg-opacity-70 backdrop-filter backdrop-blur-lg flex items-center justify-center bg-gray-800 ${customClassNames}`}
+          onClick={onCloseSliderBar}
+          onKeyDown={onCloseSliderBar}
+        >
+          {/* Content goes here */}
```
Comment (phattruong-agilityio) [inline practice-two/src/components/SliderBar.jsx]: Does this comment is helpful?

Comment (TheHienDuong) [reply practice-two/src/components/SliderBar.jsx]: Removed comment and updated associated with a text label.

File: practice-two/src/components/SliderBar.jsx (hunk 7 above)
Comment (TheHienDuong) [reply practice-two/src/components/SliderBar.jsx]: Deleted useless comments

File: practice-two/src/components/SliderBar.jsx (hunk 6 above)
Comment (TheHienDuong) [reply practice-two/src/components/SliderBar.jsx]: Yes, used the aside HTML tag to update instead of the div

**Comment by TheHienDuong** [comment]
> Why Button is inside a folder but the popover is not? image

Updated structure folder
│  ├── components
│  │  ├── ButtonFavorite
│  │  │  ├── ButtonFavorite.jsx
│  │  │  └── style.css
│  │  ├── common
│  │  │  ├── Button.jsx
│  │  │  ├── Card
│  │  │  │  ├── Card.jsx
│  │  │  │  ├── CardContent.jsx
│  │  │  │  ├── CardImage.jsx
│  │  │  │  ├── CardOverflow.jsx
│  │  │  │  └── index.js
│  │  │  ├── Chip.jsx
│  │  │  ├── Link.jsx
│  │  │  ├── Popover.jsx
│  │  │  ├── SliderBar.jsx
│  │  │  └── Typography.jsx
│  │  ├── FavoriteProductsCard.jsx
│  │  ├── GroupFilterPopover.jsx
│  │  ├── layout
│  │  │  ├── FilterCategoryLayout.jsx
│  │  │  └── ListProductsCard.jsx
│  │  ├── ManuFactureFilterPopover.jsx
│  │  ├── PricesFilterPopover.jsx
│  │  ├── ProductCard.jsx
│  │  ├── ProductDetails.jsx
│  │  └── ShoppingCart.jsx

**Pull request merged**